pygments/lexers/arturo.py

```diff
@@ -0,0 +1,312 @@
+                (words(('false', 'true', 'maybe'), suffix=r'\b'),
+                    Name.Constant)
+            ],
+            'character': [
+                (r'`.`', String.Char)
+            ],
+            'color': [
+                (r'#\w+',
+                        Name.Constant)
+            ],
+            'float': [
+                (r'[0-9]+\.[0-9]+',
+                        Number.Float)
+            ],
+            'integer': [
+                (r'[0-9]+',
+                        Number.Integer)
+            ],
+            'label': [
+                (r'\w+\b\??:',
+                        Name.Label)
+            ],
+            'literal': [
+                (r'\'(?:\w+\b\??:?)',
+                        Keyword.Declaration)
+            ],
+            'type': [
+                (r'\:\w+',
+                        Keyword.Type)
+            ],
+            'attributes': [
+                (r'\.\w+',
+                        Name.Attribute)
+            ],
+
+
+        'strings': [
+            # Single Line Strings
+            (r'"',   String.Double, 'inside-simple-string'),
+            (r'»',   String.Single, 'inside-smart-string' ),
+            (r'«««', String.Double, 'inside-safe-string'  ),
+            (r'\{\/', String.Single, 'inside-regex-string'),
+
+            # Multi Line Strings
+            (r'\{\:',   String.Double, 'inside-curly-verb-string'),
+            (r'(\{)(\!)(\w+)(\s|\n)([\w\W]*?)(^\})',
+                                         handle_annotated_strings),
+            (r'\{',     String.Single, 'inside-curly-string'     ),
+            (r'\-{3,}', String.Single, 'inside-eof-string'       ),
+        ],
+            'string-basics': [
+                include('string-interpol'),
+                include('string-escape'),
+                include('string-templates'),
+            ],
+                'string-interpol': [
+                    (r'\|',
+                        String.Interpol, 'inside-interpol'), # Interpolation
+                ],
+                    'inside-interpol': [
+                        (r'\|', String.Interpol, '#pop'),
+                        (r'.', using(this)),
+                    ],
+                'string-templates': [
+                    (r'\<\|\|',
+                        String.Interpol, 'inside-template'), # Templates
+                ],
+                    'inside-template': [
+                        (r'\|\|\>',
+                            String.Interpol, '#pop'),
+                        (r'.', using(this)),
+                    ],
+                'string-escape': [
+                    (r'\\\\', String.Escape), # Escaping backslash
+                    (r'\\n',  String.Escape), # Escaping NewLine control
+                    (r'\\t',  String.Escape), # Escaping Tabulation control
+                    (r'\\"',  String.Escape), # Escaping Quote Character
+                ],
+            'string-content-single-line': [
+                (r'.', String)
```

Comment:
again this is bad since it matches the string contents char-by-char

You can see it in the test golden output file

Reply:
- [x] [replace inefficient match](https://github.com/pygments/pygments/pull/2259/commits/7b683c48e21bb507e78ff8a549a4211455541c1d)